logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/UpdateConversationAccessRoleUseCase.kt

```diff
@@ -0,0 +1,79 @@
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.conversation.Conversation
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.functional.flatMap
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.logic.functional.map
+
+class UpdateConversationAccessRoleUseCase internal constructor(
+    private val conversationRepository: ConversationRepository
+) {
+    suspend operator fun invoke(
+        conversationId: ConversationId,
+        allowGuest: Boolean,
+        allowServices: Boolean,
+        allowNonTeamMember: Boolean
+    ): Result {
+        return conversationRepository.detailsById(conversationId)
+            .map { conversation ->
+                // TODO: handle edge case where accessRole is null
+                val newAccessRoles: List<Conversation.AccessRole>? = conversation.accessRole?.toMutableSet()?.apply {
+                    if (allowGuest) {
+                        enableGuests()
+                    } else {
+                        disableGuests()
+                    }
```

Comment:
I don't know what do you think about that but to me these methods (`enableGuests`, `disableGuests`, etc.) are redundant, they are used only in this single situation and all of them can be replaced by single-liners that are currently wrapped by these methods, or maybe I'm missing something 😄